src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,178 @@
+        public Guid UserId { get; init; }
+        public DateTime ExpiryDate { get; init; }
+        public bool CanView { get; init; }
+        public bool CanUpload { get; init; }
+        public bool CanDownload { get; init; }
+        public bool CanChangePermission { get; init; }
+    }
+    
+    public class CommandHandler : IRequestHandler<Command, EntryPermissionDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IDateTimeProvider _dateTimeProvider;
+
+        public CommandHandler(IApplicationDbContext applicationDbContext, IMapper mapper, IDateTimeProvider dateTimeProvider)
+        {
+            _context = applicationDbContext;
+            _mapper = mapper;
+            _dateTimeProvider = dateTimeProvider;
+        }
+        
+        public async Task<EntryPermissionDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var entry = await _context.Entries
+                .Include(x => x.Uploader)
+                .Include(x => x.Owner)
+                .FirstOrDefaultAsync(x => x.Id == request.EntryId, cancellationToken);
+            
+            if (entry is null)
+            {
+                throw new KeyNotFoundException("Entry does not exist.");
+            }
+
+            var hasModifyPermission = _context.EntryPermissions.Any(x =>
+                x.EntryId == request.EntryId
+                && x.EmployeeId == request.CurrentUser.Id
+                && x.AllowedOperations.Contains(EntryOperation.ChangePermission.ToString()));
+            
+            if (entry.Owner!.Id != request.CurrentUser.Id && !hasModifyPermission)
+            {
+                throw new UnauthorizedAccessException("You are not allow to change permission of this entry.");
+            }
+            
+            var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == request.UserId, cancellationToken);
+
+            if (user is null)
+            {
+                throw new KeyNotFoundException("User does not exist.");
+            }
+
+            if (user.Id == entry.OwnerId)
+            {
+                throw new ConflictException("You can not modify owner's permission.");
+            }
+
+            if (request.ExpiryDate < _dateTimeProvider.DateTimeNow)
+            {
+                throw new ConflictException("Expiry date cannot be in the past.");
+            }
+
+            var allowOperations = GenerateAllowOperations(request, entry.IsDirectory);
+            
+            await GrantPermission(entry, user, allowOperations, request.ExpiryDate, cancellationToken);
+
+            if (entry.IsDirectory)
+            {
+                // Update permissions for child Entries
+                var path = entry.Path.Equals("/") ? (entry.Path + entry.Name) : (entry.Path + "/" + entry.Name);
+                var childEntries = _context.Entries
+                    .Include(x => x.Uploader)
```

Comment:
I don't see why including these properties is a good thing. They are not used and reduce performance. This is not a premature optimization, just redundant information retrieved.

Reply:
I will remove it. It was just me not knowing what the different between `Uploader` and `Owner`.

Reply:
Fixed at https://github.com/ProFile-org/ProFile-BE/pull/336/commits/2b981a1fe890748cc93fa18e7e2a23ba1a82f683